be2-scala/src/main/scala/ch/epfl/pop/pubsub/PublishSubscribe.scala

```diff
@@ -95,9 +96,12 @@ object PublishSubscribe {
           /* partitioner port numbers */
           val portPipelineError = 0
           val portParamsWithMessage = 1
```

Comment:
nitpick: Maybe you could add separate ports for the remaining Method types to have more consistency in the ports, and then simply do `val portPublish = portParamsWithMessage ` and `val portBroadcast = portParamsWithMessage` to use them later in the partition in place of `portParamsWithMessage`

Reply:
I tried doing that but it seems to add inconsistency later on when gluing the components together since we have to use `portParamsWithMessage` anyway (line 141). It would mean that `portBroadcast` and `portPublish` would look unused, I think it makes the code less clear.

Reply:
Indeed that may look more confusing instead of helping